packages/aws-cdk-lib/aws-ec2/lib/vpc-flow-logs.ts

```diff
@@ -800,6 +835,15 @@ export class FlowLog extends FlowLogBase {
       }).join(' ');
     }
 
+    // In Transit Gateway and Transit Gateway Attachment, `trafficType` is not supported.
+    let trafficType: FlowLogTrafficType | undefined = props.trafficType ?? FlowLogTrafficType.ALL;
+    if (props.resourceType.resourceType === 'TransitGateway' || props.resourceType.resourceType === 'TransitGatewayAttachment') {
+      if (props.trafficType) {
+        throw new Error('trafficType is not supported for Transit Gateway and Transit Gateway Attachment');
+      }
+      trafficType = undefined;
+    }
+
```

Comment:
Thank you for your suggestion. 

I fully agree with moving the error handling to this location. 
(In fact, I'm embarrassed about the initial implementation where I wrote it in a very redundant place😢)

I have a question. In the implementation example you proposed, it seems that there would be an error when `maxAggregationInterval` is `undefined`. Is that acceptable? 
According to the documentation, the default value of `maxAggregationInterval` is 10 minutes, and for TransitGateway, only 1 minute can be set. However, when deploying the CFN with `maxAggregationInterval` left as undefined, the deployment proceeds normally. 

Therefore, I thought it would be acceptable to allow undefined as well. (I'm concerned about causing new errors for users who have traditionally been creating resources with maxAggregationInterval set to undefined.)

Reply:
> In the implementation example you proposed, it seems that there would be an error when maxAggregationInterval is undefined

Really? I did not envision that happening with my proposed code. I don't think the following code I just suggested would cause the error, but did it happen elsewhere?

(I think the error do not occur by `props.maxAggregationInterval && `.)

```ts
    let trafficType: FlowLogTrafficType | undefined = props.trafficType ?? FlowLogTrafficType.ALL;
    if (props.resourceType.resourceType === 'TransitGateway' || props.resourceType.resourceType === 'TransitGatewayAttachment') {
      if (props.trafficType) {
        throw new Error('trafficType is not supported for Transit Gateway and Transit Gateway Attachment');
      }
      if (props.maxAggregationInterval && props.maxAggregationInterval !== FlowLogMaxAggregationInterval.ONE_MINUTE) {
        throw new Error('maxAggregationInterval must be set to ONE_MINUTE for Transit Gateway and Transit Gateway Attachment');
      }
      trafficType = undefined;
    }
```

Reply:
@go-to-k 
oh.. I'm sorry. I overlooked `props.maxAggregationInterval`.